coral/breakpoint/breakpoint_types.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+import io
+from dataclasses import dataclass
+from typing import Any, Dict, List
+
+import intervaltree
+import numpy as np
+
+from coral.types import CnsInterval
+
+
+@dataclass
+class CNSSegData:
+    tree: intervaltree.IntervalTree
+    intervals: List[CnsInterval]
+    intervals_by_chr: Dict[str, List[List[int]]]
+    log2_cn: List[float]
+
+    @classmethod
+    def from_file(cls, file: io.TextIOWrapper) -> CNSSegData:
```

Comment:
Primarily for simplicity, since we can have click/typer pass us the file object from the CLI and do validation on it before getting to "CoRAL logic", instead of having validation embedded into functions like these as it previously was or potentially causing confusing errors while we're mid-BreakpointGraph generation. It also means we're using the same file at all points in the code, as the same ".bam" / ".cns" could be edited by some external program/user over a single CoRAL process's lifetime. https://click.palletsprojects.com/en/stable/arguments/#file-opening-safety